pkg/resource/deploy/step_generator.go

```diff
@@ -1132,10 +1132,17 @@ func diffResource(urn resource.URN, id resource.ID, oldInputs, oldOutputs,
 		return diff, err
 	}
 	if diff.Changes == plugin.DiffUnknown {
-		if oldInputs.DeepEquals(newInputs) {
```

Comment:
I'd consider adding some tests to `sdk/go/common/resource` to ensure that `PropertyValue.Diff` always returns a nil diff if `PropertyValue.DeepEquals` returns `true` and always returns a non-nil diff if `PropertyValue.Equals` returns `false`.